crates/common/tedge_config_macros/impl/src/lib.rs

```diff
@@ -61,6 +62,33 @@ pub fn generate_configuration(tokens: TokenStream) -> Result<TokenStream, syn::E
         })
         .collect::<Vec<_>>();
 
+    let fromstr_default_tests = fields_with_keys
```

Comment:
The changes in `define_tedge_config` add a new `default(from_str = "...")` attribute, which parses the given str to the default value. This section of code adds a test that explicitly checks the default value can be parsed successfully.